draft-schwenkschuster-oauth-spiffe-client-auth.md

```diff
@@ -279,32 +279,135 @@ Trust domain "example.org": Workload API at unix:///var/secrets/spiffe/agent.soc
+            5c:4b:d5:2d:f9:c1:6e:78:2c:32:a6:bb:6c:73:f0:b8:f4:be:13:09
+        Signature Algorithm: ecdsa-with-SHA512
+        Issuer: C=US, O=SPIFFE
+        Validity
+            Not Before: May 16 11:23:19 2025 GMT
+            Not After : May 15 11:23:19 2030 GMT
+        Subject: C=US, O=SPIFFE
+        Subject Public Key Info:
+            Public Key Algorithm: id-ecPublicKey
+                Public-Key: (384 bit)
+                pub:
+                    04:ef:3f:db:67:2b:e8:5c:a1:64:23:e7:f2:fd:f0:
+                    3b:16:55:68:17:55:17:d4:bd:cd:6d:04:fd:cc:8f:
+                    99:31:f7:8c:ac:b0:1e:31:60:18:45:32:8b:a1:17:
+                    4b:2f:01:75:27:6c:3f:c3:a5:b9:da:56:fb:29:54:
+                    63:cb:08:96:81:35:0e:96:04:03:40:fe:51:0d:26:
+                    da:d5:99:6c:8f:c2:45:43:cb:2c:b4:8d:9b:68:78:
+                    9f:c0:2d:68:36:b8:5e
+                ASN1 OID: secp384r1
+                NIST CURVE: P-384
+        X509v3 extensions:
+            X509v3 Subject Key Identifier:
+                8D:79:D2:26:5E:4C:83:30:40:C7:E9:1D:E1:35:12:F6:60:CF:0B:DB
+            X509v3 Basic Constraints: critical
+                CA:TRUE
+            X509v3 Key Usage: critical
+                Certificate Sign, CRL Sign
+            X509v3 Subject Alternative Name:
+                URI:spiffe://example.org
+    Signature Algorithm: ecdsa-with-SHA512
+    Signature Value:
+        30:64:02:30:0a:e9:fd:d4:cd:99:52:90:cb:14:86:93:4e:f8:
+        02:52:d6:17:12:9f:2e:65:99:0e:38:b6:b9:a6:fe:43:0f:60:
+        30:04:87:ec:24:20:80:a4:75:ee:3c:ad:9d:a2:72:0d:02:30:
+        55:93:0e:14:8c:47:47:3b:74:7c:a7:2a:2a:96:1d:a4:85:46:
+        4f:3f:95:a4:c2:ab:3c:2e:04:b3:1b:cf:02:0f:33:fc:dd:dc:
+        d5:2f:44:c8:2a:dc:ce:3f:c5:c6:89:d0
+~~~
+
+> Arndt: Bundle doesn't match X509-SVID. This needs to be fixed.
 
-The bundle endpoint is not discoverable from the JWT-SVID and X509-SVID and MUST be configured manually out of band. Bundle endpoints MUST be keyed by the trust domain identifier.
+## Alternative methods to avoid
 
-## Alternative validation methods to avoid
+The following key distribution mechanisms are alternatives and SHOULD be avoided for interopability reasons.
 
-## Manual configuration
+### SPIFFE Workload API
+
+The SPIFFE Workload API allows workloads to retrieve a trust bundle from SPIFFE. It requires the authorization server to be part of a SPIFFE trust domain and be considered a workload within it. The authors acknowledge that using the SPIFFE Workload API can reduce the time an update of the bundle is received by the authorization server.
+
+In addition to the trust bundle of the trust domain the workload resides in, the SPIFFE Workload API also allows to retrieve trust bundles from federated trust domains. This mechanism moves trust establishment away from the authorization server to the SPIFFE configuration and is NOT RECOMMENDED and more explicit configuration SHOULD be maintained.
+
```

Comment:
This seems overly restrictive - not sure about the "NOT RECOMMENDED" here - I think it comes down to the rust relationships that the authorization server accept. I can imagine that if I have both an OAuth server and a SPIFFE deployment, I would be OK to say "I trust the SPIFFE deployment, and accept all the trust relationships it establishes" - it's a risk decision that is deployment specific.

A better way to address this is to highlight the risk that a SPIFFE deployment may contain multiple federation relationships, and as a result, the OAuth deployment SHOULD make a conscious decision on whether to allow federation. The riskd ecision belongs to the deployment.